www/class/config-generate-remote/ContactGroup.php

```diff
@@ -184,7 +186,34 @@ public function getContactFromCgId(int $cgId)
     }
 
     /**
-     * Generate contact group and get contact group name
+     * Check illegal char defined into nagios.cfg file
+     *
+     * @param string $name The string to sanitize
+     * @return string The string sanitized
+     */
+    protected function checkIllegalChar($name)
+    {
+        $pollerId = $this->backendInstance->getPollerId();
+
+        if (! isset($this->illegalCharsTab[ $pollerId ])) {
+            $query = "SELECT illegal_object_name_chars FROM cfg_nagios WHERE nagios_server_id = :pollerId";
+            $stmt = $this->backendInstance->db->prepare($query);
+            $stmt->bindParam(':pollerId', $pollerId, PDO::PARAM_INT);
+            $stmt->execute();
+
+            $illegalCharsStr = implode('', $stmt->fetchAll(PDO::FETCH_COLUMN));
+            $illegalCharsStr = html_entity_decode($illegalCharsStr, ENT_QUOTES, "UTF-8");
+            $this->illegalCharsTab[ $pollerId ] = array_unique(str_split($illegalCharsStr));
+        }
+
+        if (! empty($this->illegalCharsTab[ $pollerId ])) {
+            $name = str_replace($this->illegalCharsTab[ $pollerId ], '_', $name);
+        }
+        return $name;
+    }
+
+    /**
+     * Generate contact group and get contact group name (sanitized)
```

Comment:
```suggestion
    private function sanitizeNameFromIllegalChars(string $name): string
    {
        $pollerId = $this->backendInstance->getPollerId();

        $illegalChars = $this->getIllegalCharsByNagiosServerId($pollerId);

        if (empty($illegalChars)) {
            return $name;
        }

        return str_replace($illegalChars, '_', $name);
    }

    private function getIllegalCharsByNagiosServerId(int $nagiosServerId): array
    {
        $query = 'SELECT illegal_object_name_chars FROM cfg_nagios WHERE nagios_server_id = :nagiosServerId';
        $stmt = $this->backendInstance->db->prepare($query);
        $stmt->bindParam(':nagiosServerId', $nagiosServerId, PDO::PARAM_INT);
        $stmt->execute();

        $illegalCharsStr = implode('', $stmt->fetchAll(PDO::FETCH_COLUMN));
        $illegalCharsStr = html_entity_decode($illegalCharsStr, ENT_QUOTES, 'UTF-8');

        return array_unique(str_split($illegalCharsStr));
    }
```

Reply:
The name checkIllegalChar  doesn't describe what the method is actually doing. This method retreieves illegal chars and sanitizes the string and returns them.

I propose to split it in 2 methods : a get method (from database) and a sinitize method. Then the sinitize method calls get method.

I'd make them private. no need of protected.

Reply:
> The name checkIllegalChar doesn't describe what the method is actually doing. This method retreieves illegal chars and sanitizes the string and returns them.
> 
> I propose to split it in 2 methods : a get method (from database) and a sinitize method. Then the sinitize method calls get method.
> 
> I'd make them private. no need of protected.

I'm agree with you...
If I choose a unique function named "checkIllegalChar", it's only because it's already like that in other existing files:

- www/class/centreon.class.php:291
- www/class/centreonHost.class.php:618
- www/class/centreonService.class.php:307
- www/class/centreon-clapi/centreonObject.class.php:638

So I tried to stay consistent with the existing code, I don't know which is preferable.